runtime/src/accounts_db.rs

```diff
@@ -2920,10 +2920,6 @@ impl AccountsDb {
             *accounts_hash_complete_oldest_non_ancient_slot,
```

Comment:
it is this `max` that causes us to apply it again. `max` had it applied on the previous call to this fn. Then, we get the previous value on the next call (because of the `max`)  and then apply it again on line 2925 below.

Reply:
yes, a positive offset keeps pushing it ever higher. Notably, positive offsets at the moment are ONLY used for testing when we don't have the feature enabled for disabling rewrites. So, this bug is only bad for the testing situation that exists right now.